core/src/components/checkbox/checkbox.tsx

```diff
@@ -228,6 +287,7 @@ export class Checkbox implements ComponentInterface {
             part="label"
           >
             <slot></slot>
+            {this.renderHintText()}
```

Comment:
The text (helper or error) is being read twice by VoiceOver on Safari, Firefox, and Chrome. After some quick debugging, it seems that this happening because the text is inside of label. `ion-input` has it after the `<label>` which mimics how the native input states it.

https://github.com/user-attachments/assets/d72b0a15-9266-45b7-9872-70c99b3f2562

Reply:
Tested in macOS, iOS and Android talkback and it is announcing properly after this fix: https://github.com/ionic-team/ionic-framework/pull/30140/commits/bac73833ca401338766cafcfd7480208b49b1d64